cmd/internal/labeler/labeler.go

```diff
@@ -455,8 +471,97 @@ func createNumaNodeMappingLabel(mapping map[int][]string) string {
 	return strings.Join(parts, "_")
 }
 
-func (l *labeler) printLabels() {
+func (l *labeler) printLabelsToFile(labelFile string) error {
+	f, err := os.OpenFile(labelFile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		klog.Warning("Failed to open file:", err)
+
+		return err
+	}
+
+	for key, val := range l.labels {
+		if _, err := f.WriteString(key + "=" + val + "\n"); err != nil {
+			klog.Warning("Failed to write label to file:", err)
+
+			return err
+		}
+	}
+
+	return nil
```

Comment:
File should be closed here & on errors. Close success should be also checked, or error for close reported.